docs/advanced_puppet_configuration.md

```diff
@@ -0,0 +1,14 @@
+## Advanced Puppet Configuration
+-----------
+
+The splunk_hec module also supports customizing the `fact_terminus` and `facts_cache_terminus` names in the custom routes.yaml it deploys. If you are using a different facts_terminus (ie, not PuppetDB), you will want to set that parameter.
```

Comment:
I am a bit unclear on which parameter should be set if we are not using PuppetDB. I see both `facts_terminus`and `fact_terminus`in here. Are those different parameters?

Reply:
Customers tend to add two possible parameters into `fact_terminus` section in the console. We can add more clarification on the impact. 

1.  add "splunk_hec" => this will redirect facts into splunk that means puppetdb won't have facts. We can add the notes of possible impacts in standard PE infra.

2. add both "puppetdb,splunk_hec" ( "[splunk_hec,puppetdb]")  => This is accepted by Class[Splunk_hec], but break `pe-puppetserver` service. We can add the warning message.
```
[root@pe201911 centos]# puppet agent -t
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for pe201911.pt.ms
Info: Applying configuration version '1588737375'
Notice: /Stage[main]/Splunk_hec/File[/etc/puppetlabs/puppet/splunk_hec_routes.yaml]/content:
--- /etc/puppetlabs/puppet/splunk_hec_routes.yaml	2020-04-21 06:18:02.096128704 +0000
+++ /tmp/puppet-file20200506-520-3six36	2020-05-06 03:56:26.201864532 +0000
@@ -2,5 +2,5 @@
 ---
 master:
   facts:
-    terminus: "splunk_hec"
+    terminus: "[splunk_hec,puppetdb]"
     cache: "splunk_hec"

Notice: /Stage[main]/Splunk_hec/File[/etc/puppetlabs/puppet/splunk_hec_routes.yaml]/content: content changed '{md5}d37117dff549ca54e15f5558628be85d' to '{md5}dca2bc8a609eafbca3f5b200495259b7'
Info: /Stage[main]/Splunk_hec/File[/etc/puppetlabs/puppet/splunk_hec_routes.yaml]: Scheduling refresh of Service[pe-puppetserver]
Error: /Stage[main]/Puppet_enterprise::Master::Puppetserver/Puppet_enterprise::Trapperkeeper::Pe_service[puppetserver]/Service[pe-puppetserver]: Failed to call refresh: Systemd restart for pe-puppetserver failed!
journalctl log for pe-puppetserver:
-- Logs begin at Wed 2020-04-15 04:32:14 UTC, end at Wed 2020-05-06 03:56:45 UTC. --
May 06 03:56:35 pe201911.pt.ms systemd[1]: Reloading pe-puppetserver Service.
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service: main process exited, code=exited, status=1/FAILURE
May 06 03:56:45 pe201911.pt.ms puppetserver[874]: Process 14064 exited before reload had completed
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service: control process exited, code=exited status=1
May 06 03:56:45 pe201911.pt.ms systemd[1]: Reload failed for pe-puppetserver Service.
May 06 03:56:45 pe201911.pt.ms systemd[1]: Unit pe-puppetserver.service entered failed state.
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service failed.

Error: /Stage[main]/Puppet_enterprise::Master::Puppetserver/Puppet_enterprise::Trapperkeeper::Pe_service[puppetserver]/Service[pe-puppetserver]: Systemd restart for pe-puppetserver failed!
journalctl log for pe-puppetserver:
-- Logs begin at Wed 2020-04-15 04:32:14 UTC, end at Wed 2020-05-06 03:56:45 UTC. --
May 06 03:56:35 pe201911.pt.ms systemd[1]: Reloading pe-puppetserver Service.
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service: main process exited, code=exited, status=1/FAILURE
May 06 03:56:45 pe201911.pt.ms puppetserver[874]: Process 14064 exited before reload had completed
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service: control process exited, code=exited status=1
May 06 03:56:45 pe201911.pt.ms systemd[1]: Reload failed for pe-puppetserver Service.
May 06 03:56:45 pe201911.pt.ms systemd[1]: Unit pe-puppetserver.service entered failed state.
May 06 03:56:45 pe201911.pt.ms systemd[1]: pe-puppetserver.service failed.

Error: /Stage[main]/Puppet_enterprise::Profile::Console::Proxy/File[/etc/puppetlabs/nginx/dhparam_puppetproxy.pem]: Could not evaluate: Could not retrieve file metadata for puppet:///modules/puppet_enterprise/console/dhparam_puppetproxy.pem: Failed to open TCP connection to pe201911.pt.ms:8140 (Connection refused - connect(2) for "pe201911.pt.ms" port 8140)
Notice: /Stage[main]/Pe_nginx/Service[pe-nginx]: Dependency File[/etc/puppetlabs/nginx/dhparam_puppetproxy.pem] has failures: true
Warning: /Stage[main]/Pe_nginx/Service[pe-nginx]: Skipping because of failed dependencies
Info: Stage[main]: Unscheduling all events on Stage[main]
Notice: Applied catalog in 30.84 seconds
Error: Could not send report: Failed to open TCP connection to pe201911.pt.ms:8140 (Connection refused - connect(2) for "pe201911.pt.ms" port 8140)
```

Reply:
@jarretlavallee there is just the one facts_terminus like in [](https://github.com/puppetlabs/puppetlabs-splunk_hec/blob/master/templates/splunk_hec_routes.yaml.epp)

Reply:
@mahesws for comment 1, the user would only ever want to set the facts_terminus if they are already using something other than puppetdb, in which case they would set it to whatever they are using. For comment 2, the facts cache terminus should always be set to only splunk_hec. This is safe because the splunk_hec cache does everything the default cache does and then just does the splunk stuff at the end.

Reply:
```suggestion
The splunk_hec module also supports customizing the `facts_terminus` and `facts_cache_terminus` names in the custom routes.yaml it deploys. If you are using a different facts_terminus (ie, not PuppetDB), you will want to set that parameter.
```